### PR TITLE
Feature/tracking/user properties

### PR DESCRIPTION
**Pull request by gustaveen**
## What?
Track user properties

Branched of https://github.com/HedvigInsurance/market-web/pull/598

## Why?
We need to track the same user properties in marker-web as we do in web-onboarding

Screenshot 2021-09-02 at 10 50 38

### CODE REVIEW

File: src/utils/tracking/gtm.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
+import { useEffect } from 'react'
+import { useLocale } from 'context/LocaleContext'
+export type AppEnvironment = 'development' | 'staging' | 'production'
+
+type GTMUserProperties = {
+  market: string
+  environment: AppEnvironment | undefined
```
Comment (robinandeer) [inline src/utils/tracking/gtm.ts:7]:
```suggestion
  environment?: AppEnvironment
```

File: src/server/utils/tracking.ts [hunk 2]
```diff
@@ -17,6 +17,10 @@ const snap = `
 `
 
 export const allTracking = (nonce: string) => `
+<script nonce="${nonce}">
+  dataLayer = [];
```
Comment (robinandeer) [inline src/server/utils/tracking.ts:21]: Is it more clear to write it out like:

```suggestion
  window.dataLayer = [];
```

**Review by robinandeer — DISMISSED** [review]
It's slightly confusing since you have the `useLocale` changes in here also 😅 but I think it looks good!

**Comment by gustaveen** [comment]
> It's slightly confusing since you have the `useLocale` changes in here also 😅 but I think it looks good!

Ah sorry for that :D Will merge the other PR first 🙃

File: src/server/utils/tracking.ts (hunk 2 above)
Comment (gustaveen) [reply src/server/utils/tracking.ts:21]: I kinda like to stick to the dev docs from google https://developers.google.com/tag-manager/devguide#adding-data-layer-variables-to-a-page :)

File: src/context/LocaleContext.tsx [hunk 3]
```diff
@@ -9,13 +9,7 @@ export const LocaleContext = createContext<LocaleContext>({
   currentLocale: locales['se-en'],
```
Comment (styrelius) [inline src/context/LocaleContext.tsx:9]: Sorry this is out of context for this PR but does the `LocaleContext` need to be exported?

**Review by styrelius — COMMENTED** [review]
Looks very good to me! Just one thing that's actually unrelated to this 🙈

File: src/context/LocaleContext.tsx (hunk 3 above)
Comment (gustaveen) [reply src/context/LocaleContext.tsx:9]: Oh, not at all, well spotted! 🙏

**Review by styrelius — APPROVED** [review]
🛤️ 👑 ✨

**Pull request merged**